src/core/unit.c

```diff
@@ -2611,11 +2611,10 @@ static int unit_log_resources(Unit *u) {
 finish:
         free_many_charp(message_parts, n_message_parts);
 
-        for (size_t i = 0; i < n_iovec; i++)
-                free(iovec[i].iov_base);
+        FOREACH_ARRAY(i, iovec, n_iovec)
+                free(i->iov_base);
```

Comment:
So we already have a helper iovec_array_free(), maybe we should add iovec_array_done() too, which does a shallow free loop, like this one?

Reply:
So eventually I decided to switch iovec[] to heap allocation too. It is nicer since we can use `CLEANUP_ARRAY` without worrying whether a goto is inserted correctly.